### DISCUSSION TITLE
spaCy TRF NER training log not updating live

### DISCUSSION THREAD

**Discussion by dummynov1** [explosion/spaCy#10325] [Help: Other Questions]
I'm currently trying Spacy NER training using Transformer pipeline with GPU

When training gets started the below log loads up only with heading " =====Training pipeline====" and nothing below the header.  It stays the same until the entire model training is finished and then it shows all the epoch summary together as given below. I'm facing this issue only in Jupyter.
While on Colab, i get each row (200 step) summary as soon as it gets completed and it keeps on adding results row by row.

May i know if i have to enable any feature or set any options within Jupyter notebook to get same Training live update log (row wise update) as in Colab.

`============================= Training pipeline =============================
ℹ Pipeline: ['transformer', 'ner']
ℹ Initial learn rate: 5e-05
E    #       LOSS TRANS...  LOSS NER  ENTS_F  ENTS_P  ENTS_R  SCORE 
---  ------  -------------  --------  ------  ------  ------  ------
  0       0         398.75     40.97    2.84    3.36    2.46    0.03
  0     200         906.30   1861.38   94.51   94.00   95.03    0.95
  0     400         230.06   1028.51   98.10   97.32   98.89    0.98
  0     600          90.22   1013.38   98.99   98.40   99.58    0.99
  0     800          80.64   1131.73   99.02   98.25   99.81    0.99
  0    1000          98.50   1260.47   99.50   99.16   99.85    1.00
  0    1200          73.32   1414.91   99.49   99.25   99.73    0.99
  0    1400          84.94   1529.75   99.70   99.56   99.85    1.00
  0    1600          55.61   1697.55   99.75   99.63   99.87    1.00
  0    1800          80.41   1936.64   99.75   99.63   99.87    1.00
  0    2000         115.39   2125.54   99.78   99.69   99.87    1.00
  0    2200          63.06   2395.48   99.80   99.75   99.85    1.00
  0    2400         104.14   2574.36   99.87   99.79   99.96    1.00
  0    2600          86.07   2308.35   99.88   99.79   99.97    1.00
  0    2800          81.05   1853.15   99.90   99.87   99.93    1.00
  0    3000          52.67   1462.61   99.96   99.93   99.99    1.00
  0    3200          57.99   1154.62   99.94   99.91   99.97    1.00
  0    3400         110.74    847.50   99.90   99.85   99.96    1.00
  0    3600          90.49    621.99   99.90   99.91   99.90    1.00
  0    3800          51.03    378.93   99.87   99.78   99.97    1.00
  0    4000          93.40    274.80   99.95   99.93   99.97    1.00
  0    4200         138.98    203.28   99.91   99.87   99.96    1.00
  0    4400         106.16    127.60   99.70   99.75   99.64    1.00
  0    4600          70.28     87.25   99.95   99.94   99.96    1.00
✔ Saved pipeline to output directory
training/model-last
 `

**Your Environment**
spaCy version: 3.2.2
Platform: Win 10
Python version: 3.7.10
IDE : Jupyter v1.0.0

**Comment by polm** [comment]
This sounds like it's just Python output buffering, which is not specific to spaCy. See [here](https://stackoverflow.com/questions/107705/disable-output-buffering) or [here](https://stackoverflow.com/questions/29772158/make-ipython-notebook-print-in-real-time) for workarounds.

**Reply by dummynov1** [reply]
> This sounds like it's just Python output buffering, which is not specific to spaCy. See [here](https://stackoverflow.com/questions/107705/disable-output-buffering) or [here](https://stackoverflow.com/questions/29772158/make-ipython-notebook-print-in-real-time) for workarounds.

Could it be something to do with Python 3.7.x versions.?

**Reply by polm** [reply]
No, output buffering has been in Python a long time, and I don't think there have been any changes to behavior in a long time either.

**Reply by dummynov1** [reply]
I also feel that this behavior is mostly observed while running any Terminal / CLI commands like `! python -m spacy train ....` within Jupyter notebook. Works fine for normal scripts.

**Reply by polm** [reply]
The output buffering only affects Python, so yes, that is expected.
